Formula/opencv.rb

```diff
@@ -0,0 +1,157 @@
+  option "without-python", "Build without Python support"
+
+  option :cxx11
+
+  depends_on :ant => :build if build.with? "java"
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "eigen"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended unless MacOS.version > :snow_leopard
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :python3 => :optional
+
+  with_python = build.with?("python") || build.with?("python3")
+  pythons = build.with?("python3") ? ["with-python3"] : []
+  depends_on "numpy" => [:recommended] + pythons if with_python
+
+  # Dependencies use fortran, which leads to spurious messages about gcc
+  cxxstdlib_check :skip
+
+  resource "icv-macosx" do
+    url "https://raw.githubusercontent.com/opencv/opencv_3rdparty/81a676001ca8075ada498583e4166079e5744668/ippicv/ippicv_macosx_20151201.tgz", :using => :nounzip
+    sha256 "8a067e3e026195ea3ee5cda836f25231abb95b82b7aa25f0d585dc27b06c3630"
+  end
+
+  def arg_switch(opt)
+    build.with?(opt) ? "ON" : "OFF"
+  end
+
+  def install
+    ENV.cxx11 if build.cxx11?
+
+    args = std_cmake_args + %w[
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=
+      -DBUILD_JASPER=OFF
+      -DBUILD_JPEG=ON
+      -DBUILD_OPENEXR=OFF
+      -DBUILD_PERF_TESTS=OFF
+      -DBUILD_PNG=OFF
+      -DBUILD_TESTS=OFF
+      -DBUILD_TIFF=OFF
+      -DBUILD_ZLIB=OFF
+      -DWITH_1394=OFF
+      -DWITH_CUDA=OFF
+      -DWITH_EIGEN=ON
+      -DWITH_GPHOTO2=OFF
+      -DWITH_GSTREAMER=OFF
+      -DWITH_JASPER=OFF
+      -DWITH_OPENEXR=ON
+      -DWITH_OPENGL=OFF
+      -DWITH_QT=OFF
+      -DWITH_VTK=OFF
+    ]
+
+    args << "-DBUILD_opencv_java=" + arg_switch("java")
+    args << "-DBUILD_opencv_python2=" + arg_switch("python")
+    args << "-DBUILD_opencv_python3=" + arg_switch("python3")
+    args << "-DWITH_FFMPEG=" + arg_switch("ffmpeg")
+    args << "-DWITH_TBB=" + arg_switch("tbb")
+
+    if build.with? "contrib"
+      resource("contrib").stage buildpath/"opencv_contrib"
+      args << "-DOPENCV_EXTRA_MODULES_PATH=#{buildpath}/opencv_contrib/modules"
+    end
+
+    if build.with?("python3") && build.with?("python")
+      # Opencv3 Does not support building both Python 2 and 3 versions
+      odie "opencv3: Does not support building both Python 2 and 3 wrappers"
+    end
+
+    if build.with? "python"
+      py_prefix = `python-config --prefix`.chomp
+      py_lib = "#{py_prefix}/lib"
+      args << "-DPYTHON2_EXECUTABLE=#{which "python"}"
+      args << "-DPYTHON2_LIBRARY=#{py_lib}/libpython2.7.dylib"
+      args << "-DPYTHON2_INCLUDE_DIR=#{py_prefix}/include/python2.7"
+    end
+
+    if build.with? "python3"
+      # Reset PYTHONPATH, workaround for https://github.com/Homebrew/homebrew-science/pull/4885
+      ENV["PYTHONPATH"] = ""
```

Comment:
I'd probably just put ENV.delete("PYTHONPATH") at the beginning of install and leave it at that.